lib/legacy/zstd_legacy.h

```diff
@@ -28,6 +28,31 @@ extern "C" {
 #include "zstd_v06.h"
 #include "zstd_v07.h"
 
+#ifndef ZSTD_LEGACY_SUPPORT
```

Comment:
Right, good point.  In practice this wouldn't affect much, as all the locations that include `zstd_legacy.h` check if `ZSTD_LEGACY_SUPPORT == 0` first, but I'll add a check here just to be sure.